src/dbus/server/dbus_agent.cpp

```diff
@@ -46,14 +46,13 @@ namespace DBus {
 const struct timeval           DBusAgent::kPollTimeout = {0, 0};
 constexpr std::chrono::seconds DBusAgent::kDBusWaitAllowance;
 
-DBusAgent::DBusAgent(otbr::Host::ThreadHost &aHost, Mdns::Publisher &aPublisher)
-    : mInterfaceName(aHost.GetInterfaceName())
-    , mHost(aHost)
-    , mPublisher(aPublisher)
+DBusAgent::DBusAgent(const DependentComponents &aDeps)
+    : mInterfaceName(aDeps.mHost.GetInterfaceName())
+    , mDeps(std::move(aDeps))
```

Comment:
The `std::move` here doesn't make it use move constructor because there's a `const` qualifier for `aDeps`.